src/query-builder/QueryBuilder.ts

```diff
@@ -360,12 +360,11 @@ export abstract class QueryBuilder<Entity> {
     }
 
     /**
-     * Prints sql to stdout using console.log.
+     * Logs the SQL query to the connection logger.
      */
-    printSql(): this { // TODO rename to logSql()
+    logSql() {
         const [query, parameters] = this.getQueryAndParameters();
-        this.connection.logger.logQuery(query, parameters);
-        return this;
+        return this.connection.logger.logQuery(query, parameters);
```

Comment:
It was returning `this`, please return code back. Also this is a breaking change, to keep this into the master lets add `printSql` back with deprecated mark.

Reply:
ah okay its returning a promise now... not sure if we need this method then.

Reply:
We can't add `printSql` back because it's no longer synchronous. That's why I changed the name so that existing implementations break instead of giving weird results

Reply:
okay lets simply revert it it and mark as `@deprecated`. We'll remove this method in the future versions. The point of this method was that we can chain it in the query while building it.